libsplinter/src/admin/service/event/mod.rs

```diff
@@ -12,5 +12,71 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Module containing the `AdminServiceEventStore` trait and related implementations.
+//! Defines an `AdminServiceEvent`, stored and returned by the `AdminServiceEventStore`.
+//!
+//! The public interface includes the struct [`AdminServiceEvent`].
+//!
+//! [`AdminServiceEvent`]: struct.AdminServiceEvent.html
+
 pub mod store;
+
+use std::convert::TryFrom;
+
+use crate::admin::service::messages;
+use crate::admin::store as admin_store;
+use crate::error::InvalidStateError;
+
+#[derive(Debug, PartialEq, Eq)]
+/// Representation of an `AdminServiceEvent` defined by the admin messages
+pub struct AdminServiceEvent {
+    pub event_id: i64,
+    pub event_type: EventType,
+    pub proposal: admin_store::CircuitProposal,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+/// Native representation of the `AdminServiceEvent` enum variants
+pub enum EventType {
+    ProposalSubmitted,
+    ProposalVote(Vec<u8>),
```

Comment:
Thoughts on changing this to something like to make it clear what the Vec<u8> is?
 ```
pub enum EventType {
    ProposalSubmitted,
    ProposalVote{
        requester: Vec<u8>,
    },
...

Reply:
I think that makes sense. I know there was a comment with the original `AdminServiceEvent` that the use of the included data isn't totally obvious. 👍